### ISSUE TITLE
Fine-tune arabert using run_language_modeling.py raise an error

### ISSUE THREAD

**Issue by WaelMohammedAbed** [aub-mind/arabert#15]
Hi,
Thank you for this model.
I am trying to fine-tune the model for BertForMaskedLM. I am using ['run_language_modeling.py'](https://github.com/huggingface/transformers/tree/master/examples/language-modeling), and I created an Arabic dataset that is formatted just like the one they used (the raw WikiText-2). 
I tied the code first using the "bert-base-uncased" model and it worked fine (using the dataset that I created) and this is the command that I used:

     python drive/My\ Drive/fine_tuning/arabert/run_language_modeling.py \

    --output_dir="drive/My Drive/fine_tuning/arabert/output" \

    --model_type=bert \

    --model_name_or_path="bert-base-uncased" \

    --do_train \

    --train_data_file="drive/My Drive/fine_tuning/arabert/train.txt" \

    --do_eval \

    --eval_data_file="drive/My Drive/fine_tuning/arabert/test.txt" \

    --mlm`

However, when I used the arabert model instead of the bert-base-uncased model I got this error:

`    Traceback (most recent call last):

    File "drive/My Drive/fine_tuning/arabert/run_language_modeling.py", line 281, in <module>
    main()

    File "drive/My Drive/fine_tuning/arabert/run_language_modeling.py", line 245, in main
    trainer.train(model_path=model_path)

    File "/usr/local/lib/python3.6/dist-packages/transformers/trainer.py", line 360, in train
    train_dataloader = self.get_train_dataloader()

    File "/usr/local/lib/python3.6/dist-packages/transformers/trainer.py", line 234, in 
    get_train_dataloader

    if self.args.local_rank == -1

    File "/usr/local/lib/python3.6/dist-packages/torch/utils/data/sampler.py", line 94, in __init__
    "value, but got num_samples={}".format(self.num_samples))

    ValueError: num_samples should be a positive integer value, but got num_samples=0`

I thought maybe because the dataset is not preprocessed using Farasa. so, I used the sample text that you provided as my training and test set just to examine the process but again, I got the same error.
[test.txt](https://github.com/aub-mind/arabert/files/4782514/test.txt)
[train.txt](https://github.com/aub-mind/arabert/files/4782515/train.txt)

I noted that when I use bert-base-uncased model during the process it will show me that the model architecture is "BertForMaskedLM" while using arabert shows only BertModel. therefore, I tried another approach by exporting an araBertForMaskedLM model and used it in the fine-tuning process and that didn't work either.

So, what I am doing wrong here? and, is there any example on finetuning this model.
Thanks in advance

**Comment by WaelMohammedAbed** [comment]
I found the solution to my issue [here](https://github.com/huggingface/transformers/issues/2380). The problem was in the run_language_modeling.py script. 
I changed line 106 in the code from this:

`block_size: int = field(
        default=-1,
        metadata={
            "help": "Optional input sequence length after tokenization."
            "The training dataset will be truncated in block of this size for training."
            "Default to the model max input length for single sentence inputs (take into account special tokens)."
        },
    )`

to this:
`block_size: int = field(
        default=510,
        metadata={
            "help": "Optional input sequence length after tokenization."
            "The training dataset will be truncated in block of this size for training."
            "Default to the model max input length for single sentence inputs (take into account special tokens)."
        },
    )`

Basically, only changed the default value for the Block size to 510 instead of -1. 
I think that they updated the file and AraBert build on the previous one and that's why Bert base is working fine with the file without changing this line. While Arabert doesn't.

I hope this will help someone else

**Comment by WissamAntoun** [comment]
As the solution in the issue said you need to specify the block_size, you could have done that by adding --block_size=512.

I also did some deep digging in the code and found out that in the case of `bert-base-uncased`, the code sets the `block_size` to `tokenizer.max_len`, which is an alias for `model_max_length` which if not provided is set to `max_model_input_sizes` which is set to 512 for some of the models under `PRETRAINED_POSITIONAL_EMBEDDINGS_SIZES` in https://github.com/huggingface/transformers/blob/master/src/transformers/tokenization_bert.py 
and AraBERT is not in this dict.
